extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -431,6 +476,324 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+		else if (currentRowObjectInfo.id === 6 || currentRowObjectInfo.id === 21) {
+			await this.dscPrimaryValueDropdownGroup.updateCssStyles({ 'visibility': 'visible' });
+			if (JSON.stringify(this.valueForPrimaryDropdown.values) !== JSON.stringify(this.viewInfo.dscOnOffOptions) ||
+				this.valueForPrimaryDropdown.value !== currentRowObjectInfo.valueForPrimary) {
+				await this.valueForPrimaryDropdown.updateProperties({
+					values: this.viewInfo.dscOnOffOptions
+					, value: currentRowObjectInfo.valueForPrimary
+				});
+			}
+		}
+		// DW_COMPATIBILITY_LEVEL (26) options accepts 1(Enabled) or 0(Disabled) values as primary and secondary values
+		else if (currentRowObjectInfo.id === 26) {
+			await this.showDropdownsSection(isSecondaryCheckboxChecked);
+			if (JSON.stringify(this.valueForPrimaryDropdown.values) !== JSON.stringify(this.viewInfo.dscEnableDisableOptions) ||
+				this.valueForPrimaryDropdown.value !== currentRowObjectInfo.valueForPrimary) {
+				await this.valueForPrimaryDropdown.updateProperties({
+					values: this.viewInfo.dscEnableDisableOptions
+					, value: currentRowObjectInfo.valueForPrimary
+				});
+			}
+			if (JSON.stringify(this.valueForSecondaryDropdown.values) !== JSON.stringify(this.viewInfo.dscEnableDisableOptions) ||
+				this.valueForSecondaryDropdown.value !== currentRowObjectInfo.valueForSecondary) {
+				await this.valueForSecondaryDropdown.updateProperties({
+					values: this.viewInfo.dscEnableDisableOptions
+					, value: currentRowObjectInfo.valueForSecondary
+				});
+			}
+		}
+		// All other options accepts primary and seconday values as ON/OFF/PRIMARY(only secondary)
+		else {
+			await this.showDropdownsSection(isSecondaryCheckboxChecked);
+			if (JSON.stringify(this.valueForPrimaryDropdown.values) !== JSON.stringify(this.viewInfo.dscOnOffOptions) ||
+				this.valueForPrimaryDropdown.value !== currentRowObjectInfo.valueForPrimary) {
+				await this.valueForPrimaryDropdown.updateProperties({
+					values: this.viewInfo.dscOnOffOptions
+					, value: currentRowObjectInfo.valueForPrimary
+				});
+			}
+			if (JSON.stringify(this.valueForSecondaryDropdown.values) !== JSON.stringify(this.viewInfo.dscOnOffOptions) ||
+				this.valueForSecondaryDropdown.value !== currentRowObjectInfo.valueForSecondary) {
+				await this.valueForSecondaryDropdown.updateProperties({
+					values: this.viewInfo.dscOnOffOptions
+					, value: currentRowObjectInfo.valueForSecondary
+				});
+			}
+		}
+	}
+
+	/**
+	 * Initializes primary and secondary values of Input type
+	 * @returns GroupContainer for primary value and secondary value
+	 */
+	private async initializeDscValueInputTypeSection(): Promise<azdata.GroupContainer> {
+		// Primary value
+		this.valueForPrimaryInput = this.createInputBox(localizedConstants.ValueForPrimaryColumnHeader, async (newValue) => {
+			if (this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForPrimary !== newValue) {
```

Comment:
Same comment as before about using local variables to improve readability, for both this method and the one below it. `this.objectInfo.databaseScopedConfigurations[this.currentRowId]` causes a lot of text to wrap around in the PR, making it hard to tell which line is which.

Reply:
But do we ever change what `this.objectInfo.databaseScopedConfigurations[this.currentRowId]` points to in this method? It's an object, so it'll get passed around by reference.